Yoast/javascript
# Reintroduce Italian and Russian morphology changes for 14.7

## Summary


This PR can be summarized in the following changelog entry:

* [yoastseo] Adds some irregular plural forms of Italian words to function words and morphologyData file, including specs and a function to call the exception list.
* [yoastseo] Adds a check for whether the word belongs to stemming exceptions to the Russian stemmer.
* [yoastseo] Adds an exception check for irregular Italian diminutives to the Italian stemmer.
* [yoastseo] Adds a check for exception lists of full forms and words with multiple stems to the Russian stemmer.
* [Yoast SEO Premium] Adds recognition of common irregular plural nouns (e.g., uomo - uomini) and adjectives (e.g., ampio - ampli) in Italian.
* [Yoast SEO Premium] Improves word form recognition of Russian nouns with stems that have endings similar to noun case suffixes (e.g., сиптом, гром, запах).
* [Yoast SEO Premium] Adds recognition of common irregular diminutives in Italian, such as nouns (e.g., ovetto - uovo), adjectives (e.g., dolciastro - dolce), and verbs (e.g., scribacchiare - scrivere).
* [Yoast SEO Premium] Improves keyword recognition of Russian irregular plurals.

## Relevant technical choices:

*

## Test instructions

This PR can be tested by following these steps:

* In Italian premium:
  * Insert a text of al least 300 words;
  * Make sure all of the following forms are being recognised as forms of the same word when one of them is entered in the keyphrase and others in the title/text/meta-description: amplissima, ampissime, ampissima,
  * Make sure all of the following forms are being recognised as forms of the same word when one of them is entered in the keyphrase and others in the title/text/meta-description:tempio, templi
  * Make sure all of the following forms are being recognised as forms of the same word when one of them is entered in the keyphrase and others in the title/text/meta-description: zia, zii, zie
  * Make sure all of the following forms are being recognised as forms of the same word when one of them is entered in the keyphrase and others in the title/text/meta-description: bue, buoi
  * Try entering the following words as keyphrases: scrivere , uovo, dolce. For each word, enter diminutive and non-diminutive forms of it in the text/title/meta description (scrivere -> scritto, scribacchiare; uovo -> uova, ovetto; dolce -> dolci, dolciastro). See whether they are recognised.


* In Russian Premium: 
  * Insert a text of al least 300 words;
  * Use a noun ending in "ом" or "ах" (e.g., "диплом", "агроном", "запах") as a keyphrase and use forms of those words in the title/text/meta description (e.g., "дипломами", "агроному", "запахи"). Check that the keyphrase forms are recognized by the plugin.


## Impact check

* This PR affects the following parts of the plugin, which may require extra testing:
  *

## UI changes

## Quality assurance


Fixes https://yoast.atlassian.net/browse/LIN-492
